modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js

```diff
@@ -117,6 +117,63 @@ export default function CartQuickAdd() {
 
 				const {name, productConfiguration, urls} = parentProduct;
 
+				if (product.discontinued) {
+					const replacementProduct = formattedProducts.find(
+						(productItem) => {
+							if (productItem.skus) {
+								return (
+									productItem.skus[0].id ===
+									product.replacementSkuId
+								);
+							}
```

Comment:
We won't need this block if we have the same structure for `formattedProducts` by pushing skus[0] in L93.

Please also change the name from `formattedProducts` to `formattedSkus`.